test-framework/test_framework.go

```diff
@@ -0,0 +1,226 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2022 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Based on https://github.com/wk8/go-ordered-map, Copyright Jean Rougé
+ *
+ */
+
+package test
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/onflow/cadence/runtime/ast"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/errors"
+	"github.com/onflow/cadence/runtime/interpreter"
+	"github.com/onflow/cadence/runtime/parser"
+	"github.com/onflow/cadence/runtime/sema"
+	"github.com/onflow/cadence/runtime/stdlib"
+	"github.com/onflow/cadence/runtime/tests/utils"
+)
+
+// This Provides utility methods to easily run test-scripts.
+// Example use-case:
+//   - To run all tests in a script:
+//         RunTests("source code")
+//   - To run a single test method in a script:
+//         RunTest("source code", "testMethodName")
+//
+// It is assumed that all test methods start with the 'test' prefix.
+
+const testFunctionPrefix = "test"
+
+type Results map[string]error
+
+func RunTest(script string, funcName string) error {
+	_, inter := parseCheckAndInterpret(script)
+	_, err := inter.Invoke(funcName)
+	return err
+}
+
+func RunTests(script string) Results {
+	program, inter := parseCheckAndInterpret(script)
+
+	results := make(Results)
+
+	for _, funcDecl := range program.FunctionDeclarations() {
+		funcName := funcDecl.Identifier.Identifier
+
+		if strings.HasPrefix(funcName, testFunctionPrefix) {
+			_, err := inter.Invoke(funcName)
+			results[funcName] = err
+		}
+	}
+
+	return results
+}
+
+func parseCheckAndInterpret(script string) (*ast.Program, *interpreter.Interpreter) {
+	program, err := parser.ParseProgram(script, nil)
+
+	checker, err := newChecker(program)
+	if err != nil {
+		panic(err)
+	}
+
+	err = checker.Check()
+	if err != nil {
+		panic(err)
+	}
+
+	// TODO: validate test function signature
+	//   e.g: no return values, no arguments, etc.
```

Comment:
Maybe it would later be useful to accept a string argument as the test's name/identifier.